Content.Server/Remotes/DoorRemoteSystem.cs

```diff
@@ -0,0 +1,92 @@
+using Content.Shared.Doors.Components;
+using Content.Shared.Doors.Systems;
+using Content.Shared.Physics;
+using Content.Shared.Access.Components;
+using Content.Server.Doors.Systems;
+using Content.Server.Doors.Components;
+
+namespace Content.Server.Remotes
+{
+    public sealed class DoorRemoteSystem : EntitySystem
+    {
+        [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
+        [Dependency] private readonly SharedDoorSystem _sharedDoorSystem = default!;
+        [Dependency] private readonly DoorSystem _doorSystem = default!;
+        [Dependency] private readonly SharedInteractionSystem _interactionSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DoorRemoteComponent, UseInHandEvent>(OnInHandActivation);
+            SubscribeLocalEvent<DoorRemoteComponent, AfterInteractEvent>(OnAfterInteract);
+        }
+
+        public void OnInHandActivation(EntityUid user, DoorRemoteComponent component, UseInHandEvent args)
+        {
+            switch (component.Mode)
+            {
+                case DoorRemoteComponent.OperatingMode.OpenClose:
+                    component.Mode = DoorRemoteComponent.OperatingMode.ToggleBolts;
+                    _popupSystem.PopupEntity(Loc.GetString("door-remote-switch-state-toggle-bolts"), args.User, Filter.Entities(args.User));
+                    break;
+                case DoorRemoteComponent.OperatingMode.ToggleBolts:
+                    component.Mode = DoorRemoteComponent.OperatingMode.OpenClose; // TODO: Swítch to ToggleEmergencyAcces when EA is implemented
+                    _popupSystem.PopupEntity(Loc.GetString("door-remote-switch-state-open-close"), args.User, Filter.Entities(args.User)); // TODO: See the above comment
+                    break;
+            /*
+                case DoorRemoteComponent.OperatingMode.ToggleEmergencyAccess:
+                    component.Mode = DoorRemoteComponent.OperatingMode.OpenClose;
+                    _popupSystem.PopupEntity(Loc.GetString("door-remote-switch-state-open-close"), args.User, Filter.Entities(args.User));
+                    break;
+            */
+            }
+        }
+        private void OnAfterInteract(EntityUid uid, DoorRemoteComponent component, AfterInteractEvent args)
+        {
+            if (args.Handled
+                || args.Target == null
+                || !EntityManager.TryGetComponent<DoorComponent>(args.Target, out var doorComponent) // If it isn't a door we don't use it
+                || !EntityManager.HasComponent<AccessReaderComponent>(args.Target) // Remotes do not work on doors without access requirements
+                || !EntityManager.TryGetComponent<AirlockComponent>(args.Target, out var airlockComponent) // Remotes only work on airlocks
+                || !_interactionSystem.InRangeUnobstructed(args.User, doorComponent.Owner, -1f, CollisionGroup.Opaque))
```

Comment:
So firstly, I think this is mean to check remote -> door, or user -> door, not user -> remote.

Secondly, there's also a dedicated `ExamineSystemShared.InRangeUnOccluded()`. that should probably be a bit faster.

Reply:
Unfortunately `InRangeUnoccluded()` suffers from the same problem as `InRangeUnobstructed()` suffered from.
Namely lacking a check that range > 0 before this:
https://github.com/space-wizards/space-station-14/blob/c36d28f6a46be85f5ab12c29c301fb3944470f3b/Content.Shared/Examine/ExamineSystemShared.cs#L122

Reply:
I guess its not strictly necessary then. but if #6440 happens to get merged quickly you could update this.